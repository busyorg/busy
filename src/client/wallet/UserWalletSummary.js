import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedNumber } from 'react-intl';
import formatter from '../helpers/steemitFormatter';
import { calculateTotalDelegatedSP, calculateEstAccountValue } from '../vendor/steemitHelpers';
import BTooltip from '../components/BTooltip';
import Loading from '../components/Icon/Loading';
import USDDisplay from '../components/Utils/USDDisplay';
import './UserWalletSummary.less';

const getFormattedTotalDelegatedSP = (user, totalVestingShares, totalVestingFundSteem) => {
  const totalDelegatedSP = calculateTotalDelegatedSP(
    user,
    totalVestingShares,
    totalVestingFundSteem,
  );

  if (totalDelegatedSP !== 0) {
    return (
      <BTooltip
        title={
          <span>
            <FormattedMessage
              id="steem_power_delegated_to_account_tooltip"
              defaultMessage="Steem Power delegated to this account"
            />
          </span>
        }
      >
        <span>
          {totalDelegatedSP > 0 ? '(+' : '('}
          <FormattedNumber
            value={calculateTotalDelegatedSP(user, totalVestingShares, totalVestingFundSteem)}
          />
          {' SP)'}
        </span>
      </BTooltip>
    );
  }

  return null;
};

const UserWalletSummary = ({
  user,
  loading,
  totalVestingShares,
  totalVestingFundSteem,
  loadingGlobalProperties,
  steemRate,
  sbdRate,
  steemRateLoading,
}) => (
  <div className="UserWalletSummary">
    <div className="UserWalletSummary__item">
      <i className="iconfont icon-steem UserWalletSummary__icon" />
      <div className="UserWalletSummary__label">
        <FormattedMessage id="steem" defaultMessage="Steem" />
      </div>
      <div className="UserWalletSummary__value">
        {loading ? (
          <Loading />
        ) : (
          <span>
            <FormattedNumber value={parseFloat(user.balance)} />
            {' STEEM'}
          </span>
        )}
      </div>
    </div>
    <div className="UserWalletSummary__item">
      <i className="iconfont icon-flashlight_fill UserWalletSummary__icon" />
      <div className="UserWalletSummary__label">
        <FormattedMessage id="steem_power" defaultMessage="Steem Power" />
      </div>
      <div className="UserWalletSummary__value">
        {loading || loadingGlobalProperties ? (
          <Loading />
        ) : (
          <span>
            <FormattedNumber
              value={parseFloat(
                formatter.vestToSteem(
                  user.vesting_shares,
                  totalVestingShares,
                  totalVestingFundSteem,
                ),
              )}
            />
            {' SP '}
            {getFormattedTotalDelegatedSP(user, totalVestingShares, totalVestingFundSteem)}
          </span>
        )}
      </div>
    </div>
    <div className="UserWalletSummary__item">
      <i className="iconfont icon-Dollar UserWalletSummary__icon" />
      <div className="UserWalletSummary__label">
        <FormattedMessage id="steem_dollar" defaultMessage="Steem Dollar" />
      </div>
      <div className="UserWalletSummary__value">
        {loading ? (
          <Loading />
        ) : (
          <span>
            <FormattedNumber value={parseFloat(user.sbd_balance)} />
            {' SBD'}
          </span>
        )}
      </div>
    </div>
    <div className="UserWalletSummary__item">
      <i className="iconfont icon-savings UserWalletSummary__icon" />
      <div className="UserWalletSummary__label">
        <FormattedMessage id="savings" defaultMessage="Savings" />
      </div>
      <div className="UserWalletSummary__value">
        {loading ? (
          <Loading />
        ) : (
          <span>
            <FormattedNumber value={parseFloat(user.savings_balance)} />
            {' STEEM, '}
            <FormattedNumber value={parseFloat(user.savings_sbd_balance)} />
            {' SBD'}
          </span>
        )}
      </div>
    </div>
    <div className="UserWalletSummary__item">
      <i className="iconfont icon-people_fill UserWalletSummary__icon" />
      <div className="UserWalletSummary__label">
        <FormattedMessage id="est_account_value" defaultMessage="Est. Account Value" />
      </div>
      <div className="UserWalletSummary__value">
        {loading || loadingGlobalProperties || steemRateLoading ? (
          <Loading />
        ) : (
          <USDDisplay
            value={calculateEstAccountValue(
              user,
              totalVestingShares,
              totalVestingFundSteem,
              steemRate,
              sbdRate,
            )}
          />
        )}
      </div>
    </div>
  </div>
);

UserWalletSummary.propTypes = {
  loadingGlobalProperties: PropTypes.bool.isRequired,
  user: PropTypes.shape().isRequired,
  totalVestingShares: PropTypes.string.isRequired,
  totalVestingFundSteem: PropTypes.string.isRequired,
  steemRate: PropTypes.number,
  sbdRate: PropTypes.number,
  loading: PropTypes.bool,
  steemRateLoading: PropTypes.bool,
};

UserWalletSummary.defaultProps = {
  steemRate: 1,
  sbdRate: 1,
  loading: false,
  steemRateLoading: false,
};

export default UserWalletSummary;
