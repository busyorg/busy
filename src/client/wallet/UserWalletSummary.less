@import (reference) '../styles/custom.less';
@import (reference) '../styles/modules/variables.less';

.UserWalletSummary {
  border: 1px solid @white-gainsboro;
  border-right: 0;
  border-left: 0;
  border-radius: 0;
  background-color: @white;

  &__item {
    display: flex;
    border-bottom: 1px solid @white-gainsboro;
    align-items: center;
    margin: 0 15px;
    flex-direction: row;
    padding: 10px 0;

    &:last-of-type {
      border-bottom: 0;
    }
  }

  &__icon {
    color: @blue-baliHai;
    margin-right: 10px;
  }

  &__label {
    align-items: center;
    color: @black;
    display: flex;
    font-size: 18px;
  }

  &__value {
    align-items: center;
    color: @purple;
    display: flex;
    font-size: 18px;
    font-weight: bold;
    margin-left: auto;
    text-align: right;
  }

  &__reward {
    &:after {
      color: @blue-botticelli;
      content: ' - ';
    }

    &:last-of-type {
      &:after {
        content: '';
      }
    }
  }

  @media @small {
    border: 1px solid @white-gainsboro;
    border-radius: @border-radius-base;
  }
}

div[data-dir='rtl'] .UserWalletSummary {
  &__icon {
    margin-right: 0;
    margin-left: 10px;
  }

  &__value {
    direction: ltr;
    margin-left: 0;
    margin-right: auto;
    text-align: left;
  }
}
