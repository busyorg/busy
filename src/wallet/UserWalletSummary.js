import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedNumber } from 'react-intl';
import steem from 'steem';
import { Tooltip } from 'antd';
import { calculateTotalDelegatedSP } from '../vendor/steemitHelpers';
import Loading from '../components/Icon/Loading';
import USDDisplay from '../components/Utils/USDDisplay';
import './UserWalletSummary.less';

const getFormattedTotalDelegatedSP = (user, totalVestingShares, totalVestingFundSteem) => {
  const totalDelegatedSP = calculateTotalDelegatedSP(
    user,
    totalVestingShares,
    totalVestingFundSteem,
  );

  if (totalDelegatedSP !== 0) {
    return (
      <Tooltip
        title={
          <span>
            <FormattedMessage
              id="steem_power_delegated_to_account_tooltip"
              defaultMessage="Steem Power delegated to this account"
            />
          </span>
        }
      >
        <span>
          {totalDelegatedSP > 0 ? '(+' : '('}
          <FormattedNumber
            value={calculateTotalDelegatedSP(user, totalVestingShares, totalVestingFundSteem)}
          />
          {' SP)'}
        </span>
      </Tooltip>
    );
  }

  return null;
};

const UserWalletSummary = ({
  user,
  estAccountValue,
  loading,
  totalVestingShares,
  totalVestingFundSteem,
  loadingEstAccountValue,
  loadingGlobalProperties,
}) => (
  <div className="UserWalletSummary">
    <div className="UserWalletSummary__item">
      <i className="iconfont icon-steem UserWalletSummary__icon" />
      <div className="UserWalletSummary__label">
        <FormattedMessage id="steem" defaultMessage="Steem" />
      </div>
      <div className="UserWalletSummary__value">
        {loading
          ? <Loading />
          : <span><FormattedNumber value={parseFloat(user.balance)} />{' STEEM'}</span>}
      </div>
    </div>
    <div className="UserWalletSummary__item">
      <i className="iconfont icon-flashlight_fill UserWalletSummary__icon" />
      <div className="UserWalletSummary__label">
        <FormattedMessage id="steem_power" defaultMessage="Steem Power" />
      </div>
      <div className="UserWalletSummary__value">
        {loading || loadingGlobalProperties
          ? <Loading />
          : <span>
            <FormattedNumber
              value={parseFloat(
                steem.formatter.vestToSteem(
                  user.vesting_shares,
                  totalVestingShares,
                  totalVestingFundSteem,
                ),
              )}
            />
            {' SP '}
            {getFormattedTotalDelegatedSP(user, totalVestingShares, totalVestingFundSteem)}
          </span>}
      </div>
    </div>
    <div className="UserWalletSummary__item">
      <i className="iconfont icon-Dollar UserWalletSummary__icon" />
      <div className="UserWalletSummary__label">
        <FormattedMessage id="steem_dollar" defaultMessage="Steem Dollar" />
      </div>
      <div className="UserWalletSummary__value">
        {loading
          ? <Loading />
          : <span><FormattedNumber value={parseFloat(user.sbd_balance)} />{' SBD'}</span>}
      </div>
    </div>
    <div className="UserWalletSummary__item">
      <i className="iconfont icon-savings UserWalletSummary__icon" />
      <div className="UserWalletSummary__label">
        <FormattedMessage id="savings" defaultMessage="Savings" />
      </div>
      <div className="UserWalletSummary__value">
        {loading
          ? <Loading />
          : <span>
            <FormattedNumber value={parseFloat(user.savings_balance)} />{' STEEM, '}
            <FormattedNumber value={parseFloat(user.savings_sbd_balance)} />{' SBD'}
          </span>}
      </div>
    </div>
    <div className="UserWalletSummary__item">
      <i className="iconfont icon-people_fill UserWalletSummary__icon" />
      <div className="UserWalletSummary__label">
        <FormattedMessage id="est_account_value" defaultMessage="Est. Account Value" />
      </div>
      <div className="UserWalletSummary__value">
        {loadingEstAccountValue || !estAccountValue
          ? <Loading />
          : <USDDisplay value={parseFloat(estAccountValue)} />}
      </div>
    </div>
  </div>
);

UserWalletSummary.propTypes = {
  user: PropTypes.shape().isRequired,
  estAccountValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  totalVestingShares: PropTypes.string.isRequired,
  totalVestingFundSteem: PropTypes.string.isRequired,
  loading: PropTypes.bool,
  loadingEstAccountValue: PropTypes.bool,
  loadingGlobalProperties: PropTypes.bool.isRequired,
};

UserWalletSummary.defaultProps = {
  loading: false,
  loadingEstAccountValue: false,
  estAccountValue: null,
};

export default UserWalletSummary;
