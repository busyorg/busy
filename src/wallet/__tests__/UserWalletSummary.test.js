import React from 'react';
import { shallow } from 'enzyme';
import UserWalletSummary from '../UserWalletSummary';

describe('(Component) UserWalletSummary', () => {
  describe('with default prop values', () => {
    it('renders and matches snapshot', () => {
      const props = {
        user: {
          balance: '100',
          vesting_shares: '0',
          savings_balance: '100 STEEM',
          savings_sbd_balance: '1000 STEEM',
        },
        estAccountValue: '100.00 STEEM',
        totalVestingShares: '100 STEEM',
        totalVestingFundSteem: '100 STEEM',
        loading: false,
      };
      const wrapper = shallow(<UserWalletSummary {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
