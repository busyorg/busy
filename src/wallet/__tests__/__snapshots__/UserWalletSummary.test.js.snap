// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) UserWalletSummary with default prop values renders and matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="UserWalletSummary"
  >
    <div
      className="UserWalletSummary__item"
    >
      <i
        className="iconfont icon-steem UserWalletSummary__icon"
      />
      <div
        className="UserWalletSummary__label"
      >
        <FormattedMessage
          defaultMessage="Steem"
          id="steem"
          values={Object {}}
        />
      </div>
      <div
        className="UserWalletSummary__value"
      >
        <span>
          <FormattedNumber
            value={100}
          />
           STEEM
        </span>
      </div>
    </div>
    <div
      className="UserWalletSummary__item"
    >
      <i
        className="iconfont icon-flashlight_fill UserWalletSummary__icon"
      />
      <div
        className="UserWalletSummary__label"
      >
        <FormattedMessage
          defaultMessage="Steem Power"
          id="steem_power"
          values={Object {}}
        />
      </div>
      <div
        className="UserWalletSummary__value"
      >
        <span>
          <FormattedNumber
            value={0}
          />
           SP 
          <Tooltip
            arrowPointAtCenter={false}
            autoAdjustOverflow={true}
            mouseEnterDelay={0.1}
            mouseLeaveDelay={0.1}
            placement="top"
            prefixCls="ant-tooltip"
            title={
              <span>
                <FormattedMessage
                  defaultMessage="Steem Power delegated to this account"
                  id="steem_power_delegated_to_account_tooltip"
                  values={Object {}}
                />
              </span>
            }
            transitionName="zoom-big-fast"
          >
            <span>
              (
              <FormattedNumber
                value={NaN}
              />
               SP)
            </span>
          </Tooltip>
        </span>
      </div>
    </div>
    <div
      className="UserWalletSummary__item"
    >
      <i
        className="iconfont icon-Dollar UserWalletSummary__icon"
      />
      <div
        className="UserWalletSummary__label"
      >
        <FormattedMessage
          defaultMessage="Steem Dollar"
          id="steem_dollar"
          values={Object {}}
        />
      </div>
      <div
        className="UserWalletSummary__value"
      >
        <span>
          <FormattedNumber
            value={NaN}
          />
           SBD
        </span>
      </div>
    </div>
    <div
      className="UserWalletSummary__item"
    >
      <i
        className="iconfont icon-savings UserWalletSummary__icon"
      />
      <div
        className="UserWalletSummary__label"
      >
        <FormattedMessage
          defaultMessage="Savings"
          id="savings"
          values={Object {}}
        />
      </div>
      <div
        className="UserWalletSummary__value"
      >
        <span>
          <FormattedNumber
            value={100}
          />
           STEEM, 
          <FormattedNumber
            value={1000}
          />
           SBD
        </span>
      </div>
    </div>
    <div
      className="UserWalletSummary__item"
    >
      <i
        className="iconfont icon-people_fill UserWalletSummary__icon"
      />
      <div
        className="UserWalletSummary__label"
      >
        <FormattedMessage
          defaultMessage="Est. Account Value"
          id="est_account_value"
          values={Object {}}
        />
      </div>
      <div
        className="UserWalletSummary__value"
      >
        <USDDisplay
          value={100}
        />
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="UserWalletSummary"
    >
      <div
        className="UserWalletSummary__item"
      >
        <i
          className="iconfont icon-steem UserWalletSummary__icon"
        />
        <div
          className="UserWalletSummary__label"
        >
          <FormattedMessage
            defaultMessage="Steem"
            id="steem"
            values={Object {}}
          />
        </div>
        <div
          className="UserWalletSummary__value"
        >
          <span>
            <FormattedNumber
              value={100}
            />
             STEEM
          </span>
        </div>
      </div>
      <div
        className="UserWalletSummary__item"
      >
        <i
          className="iconfont icon-flashlight_fill UserWalletSummary__icon"
        />
        <div
          className="UserWalletSummary__label"
        >
          <FormattedMessage
            defaultMessage="Steem Power"
            id="steem_power"
            values={Object {}}
          />
        </div>
        <div
          className="UserWalletSummary__value"
        >
          <span>
            <FormattedNumber
              value={0}
            />
             SP 
            <Tooltip
              arrowPointAtCenter={false}
              autoAdjustOverflow={true}
              mouseEnterDelay={0.1}
              mouseLeaveDelay={0.1}
              placement="top"
              prefixCls="ant-tooltip"
              title={
                <span>
                  <FormattedMessage
                    defaultMessage="Steem Power delegated to this account"
                    id="steem_power_delegated_to_account_tooltip"
                    values={Object {}}
                  />
                </span>
              }
              transitionName="zoom-big-fast"
            >
              <span>
                (
                <FormattedNumber
                  value={NaN}
                />
                 SP)
              </span>
            </Tooltip>
          </span>
        </div>
      </div>
      <div
        className="UserWalletSummary__item"
      >
        <i
          className="iconfont icon-Dollar UserWalletSummary__icon"
        />
        <div
          className="UserWalletSummary__label"
        >
          <FormattedMessage
            defaultMessage="Steem Dollar"
            id="steem_dollar"
            values={Object {}}
          />
        </div>
        <div
          className="UserWalletSummary__value"
        >
          <span>
            <FormattedNumber
              value={NaN}
            />
             SBD
          </span>
        </div>
      </div>
      <div
        className="UserWalletSummary__item"
      >
        <i
          className="iconfont icon-savings UserWalletSummary__icon"
        />
        <div
          className="UserWalletSummary__label"
        >
          <FormattedMessage
            defaultMessage="Savings"
            id="savings"
            values={Object {}}
          />
        </div>
        <div
          className="UserWalletSummary__value"
        >
          <span>
            <FormattedNumber
              value={100}
            />
             STEEM, 
            <FormattedNumber
              value={1000}
            />
             SBD
          </span>
        </div>
      </div>
      <div
        className="UserWalletSummary__item"
      >
        <i
          className="iconfont icon-people_fill UserWalletSummary__icon"
        />
        <div
          className="UserWalletSummary__label"
        >
          <FormattedMessage
            defaultMessage="Est. Account Value"
            id="est_account_value"
            values={Object {}}
          />
        </div>
        <div
          className="UserWalletSummary__value"
        >
          <USDDisplay
            value={100}
          />
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <UserWalletSummary
        estAccountValue="100.00 STEEM"
        loading={false}
        loadingEstAccountValue={false}
        totalVestingFundSteem="100 STEEM"
        totalVestingShares="100 STEEM"
        user={
          Object {
            "balance": "100",
            "savings_balance": "100 STEEM",
            "savings_sbd_balance": "1000 STEEM",
            "vesting_shares": "0",
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "estAccountValue": "100.00 STEEM",
          "loading": false,
          "loadingEstAccountValue": false,
          "totalVestingFundSteem": "100 STEEM",
          "totalVestingShares": "100 STEEM",
          "user": Object {
            "balance": "100",
            "savings_balance": "100 STEEM",
            "savings_sbd_balance": "1000 STEEM",
            "vesting_shares": "0",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="UserWalletSummary"
        >
          <div
            className="UserWalletSummary__item"
          >
            <i
              className="iconfont icon-steem UserWalletSummary__icon"
            />
            <div
              className="UserWalletSummary__label"
            >
              <FormattedMessage
                defaultMessage="Steem"
                id="steem"
                values={Object {}}
              />
            </div>
            <div
              className="UserWalletSummary__value"
            >
              <span>
                <FormattedNumber
                  value={100}
                />
                 STEEM
              </span>
            </div>
          </div>
          <div
            className="UserWalletSummary__item"
          >
            <i
              className="iconfont icon-flashlight_fill UserWalletSummary__icon"
            />
            <div
              className="UserWalletSummary__label"
            >
              <FormattedMessage
                defaultMessage="Steem Power"
                id="steem_power"
                values={Object {}}
              />
            </div>
            <div
              className="UserWalletSummary__value"
            >
              <span>
                <FormattedNumber
                  value={0}
                />
                 SP 
                <Tooltip
                  arrowPointAtCenter={false}
                  autoAdjustOverflow={true}
                  mouseEnterDelay={0.1}
                  mouseLeaveDelay={0.1}
                  placement="top"
                  prefixCls="ant-tooltip"
                  title={
                    <span>
                      <FormattedMessage
                        defaultMessage="Steem Power delegated to this account"
                        id="steem_power_delegated_to_account_tooltip"
                        values={Object {}}
                      />
                    </span>
                  }
                  transitionName="zoom-big-fast"
                >
                  <span>
                    (
                    <FormattedNumber
                      value={NaN}
                    />
                     SP)
                  </span>
                </Tooltip>
              </span>
            </div>
          </div>
          <div
            className="UserWalletSummary__item"
          >
            <i
              className="iconfont icon-Dollar UserWalletSummary__icon"
            />
            <div
              className="UserWalletSummary__label"
            >
              <FormattedMessage
                defaultMessage="Steem Dollar"
                id="steem_dollar"
                values={Object {}}
              />
            </div>
            <div
              className="UserWalletSummary__value"
            >
              <span>
                <FormattedNumber
                  value={NaN}
                />
                 SBD
              </span>
            </div>
          </div>
          <div
            className="UserWalletSummary__item"
          >
            <i
              className="iconfont icon-savings UserWalletSummary__icon"
            />
            <div
              className="UserWalletSummary__label"
            >
              <FormattedMessage
                defaultMessage="Savings"
                id="savings"
                values={Object {}}
              />
            </div>
            <div
              className="UserWalletSummary__value"
            >
              <span>
                <FormattedNumber
                  value={100}
                />
                 STEEM, 
                <FormattedNumber
                  value={1000}
                />
                 SBD
              </span>
            </div>
          </div>
          <div
            className="UserWalletSummary__item"
          >
            <i
              className="iconfont icon-people_fill UserWalletSummary__icon"
            />
            <div
              className="UserWalletSummary__label"
            >
              <FormattedMessage
                defaultMessage="Est. Account Value"
                id="est_account_value"
                values={Object {}}
              />
            </div>
            <div
              className="UserWalletSummary__value"
            >
              <USDDisplay
                value={100}
              />
            </div>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="UserWalletSummary"
        >
          <div
            className="UserWalletSummary__item"
          >
            <i
              className="iconfont icon-steem UserWalletSummary__icon"
            />
            <div
              className="UserWalletSummary__label"
            >
              <FormattedMessage
                defaultMessage="Steem"
                id="steem"
                values={Object {}}
              />
            </div>
            <div
              className="UserWalletSummary__value"
            >
              <span>
                <FormattedNumber
                  value={100}
                />
                 STEEM
              </span>
            </div>
          </div>
          <div
            className="UserWalletSummary__item"
          >
            <i
              className="iconfont icon-flashlight_fill UserWalletSummary__icon"
            />
            <div
              className="UserWalletSummary__label"
            >
              <FormattedMessage
                defaultMessage="Steem Power"
                id="steem_power"
                values={Object {}}
              />
            </div>
            <div
              className="UserWalletSummary__value"
            >
              <span>
                <FormattedNumber
                  value={0}
                />
                 SP 
                <Tooltip
                  arrowPointAtCenter={false}
                  autoAdjustOverflow={true}
                  mouseEnterDelay={0.1}
                  mouseLeaveDelay={0.1}
                  placement="top"
                  prefixCls="ant-tooltip"
                  title={
                    <span>
                      <FormattedMessage
                        defaultMessage="Steem Power delegated to this account"
                        id="steem_power_delegated_to_account_tooltip"
                        values={Object {}}
                      />
                    </span>
                  }
                  transitionName="zoom-big-fast"
                >
                  <span>
                    (
                    <FormattedNumber
                      value={NaN}
                    />
                     SP)
                  </span>
                </Tooltip>
              </span>
            </div>
          </div>
          <div
            className="UserWalletSummary__item"
          >
            <i
              className="iconfont icon-Dollar UserWalletSummary__icon"
            />
            <div
              className="UserWalletSummary__label"
            >
              <FormattedMessage
                defaultMessage="Steem Dollar"
                id="steem_dollar"
                values={Object {}}
              />
            </div>
            <div
              className="UserWalletSummary__value"
            >
              <span>
                <FormattedNumber
                  value={NaN}
                />
                 SBD
              </span>
            </div>
          </div>
          <div
            className="UserWalletSummary__item"
          >
            <i
              className="iconfont icon-savings UserWalletSummary__icon"
            />
            <div
              className="UserWalletSummary__label"
            >
              <FormattedMessage
                defaultMessage="Savings"
                id="savings"
                values={Object {}}
              />
            </div>
            <div
              className="UserWalletSummary__value"
            >
              <span>
                <FormattedNumber
                  value={100}
                />
                 STEEM, 
                <FormattedNumber
                  value={1000}
                />
                 SBD
              </span>
            </div>
          </div>
          <div
            className="UserWalletSummary__item"
          >
            <i
              className="iconfont icon-people_fill UserWalletSummary__icon"
            />
            <div
              className="UserWalletSummary__label"
            >
              <FormattedMessage
                defaultMessage="Est. Account Value"
                id="est_account_value"
                values={Object {}}
              />
            </div>
            <div
              className="UserWalletSummary__value"
            >
              <USDDisplay
                value={100}
              />
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserWalletSummary
    estAccountValue="100.00 STEEM"
    loading={false}
    loadingEstAccountValue={false}
    totalVestingFundSteem="100 STEEM"
    totalVestingShares="100 STEEM"
    user={
      Object {
        "balance": "100",
        "savings_balance": "100 STEEM",
        "savings_sbd_balance": "1000 STEEM",
        "vesting_shares": "0",
      }
    }
  />,
}
`;
